### ISSUE TITLE
Calling Stored Proc with Out Parm that returns null

### ISSUE THREAD

**Issue by tzivias** [mattn/go-oci8#219]
I am trying to call a stored procedure that has an out parameter, I found your example and your code and its working fine, until the out param is return null.  I tried setting the calling variable to datatype sql.NullString but I am still getting an oracle error of  ORA-01405: fetched column value is NULL.  Can you let me know if this doesn't work if the parameter might be null?

var er string
	_, errd := db.Exec( "call P(:1,:p_out )","2", sql.Named("p_out", er))
	
	if errd != nil {
	panic(errd)
	}

**Comment by mattn** [comment]
Did you look this code?

https://github.com/mattn/go-oci8/blob/e93e9d0c9702cb2bfad8b9ba2c4216b360ab4797/_example/dbms_output/main.go#L27-L29

**Comment by tzivias** [comment]
Yes, I did, and copied it and it worked.   I wanted to know if the out parameter happens to be null, would that work?

**Comment by mattn** [comment]
Could you please show me your code of procedure?

**Comment by tzivias** [comment]
PROCEDURE p (
      p_string                   IN       VARCHAR2,
      p_out out varchar2
   ) IS
      l_string   LONG DEFAULT p_string;
   BEGIN
      LOOP
         EXIT WHEN l_string IS NULL;
         DBMS_OUTPUT.PUT_LINE (SUBSTR (l_string, 1, 250));
         l_string := SUBSTR (l_string, 251);
         p_out := 'hi';
      END LOOP;
   END p;

when I commented out    p_out := 'hi';  it gave me that error.

**Comment by ckanner** [comment]
#220 Anybody can fix it?

**Comment by kurt-google** [comment]
This is caused by the bind calls specifying `nil` for the indicator variable. Call sites like
https://github.com/mattn/go-oci8/blob/048cd9670c0a15373c1b367e77359f2a3b95a468/statement.go#L232

Its a bit vague, but when the indicator is not returned to oracle an error will be reported to the user to allow capture of these potentially error prone states. per https://docs.oracle.com/cd/B10500_01/appdev.920/a96584/oci02bas.htm#422173 and https://docs.oracle.com/cd/B28359_01/appdev.111/b28427/pc_04dat.htm#i12463

A simple, though potentially troublesome, fix is to simply pass an indicator variable (any C.sb2 will work) to the binding functions and then not use it. Im not familiar enough with the go-oci8 codebase to say how it should be handled or if it is already handled and such a fix is enough. In my usecase I do not care about null or truncated results so I am fine with passing a dummy indicator value to the functions.

**Comment by MichaelS11** [comment]
Note that Go string cannot be null, so if binding to a string, best effort is to return an empty string.

Is this still an issue otherwise?

**Comment by kurt-google** [comment]
Oracle packages are not under our control, so I cannot control what it returns.

In this case the package in error is `DMBS_OUTPUT.GET_LINE` which is provided in the example code. When called with an empty buffer it returns an error instead of appropriately binding and returning `status`=1

**Comment by MichaelS11** [comment]
Did you try this after the recent commit? This might be fixed.

**Comment by kurt-google** [comment]
Unfortunately I am on a rather old version, I will see if I can update and test. But if calling DBMS_OUTPUT.GET_LINE on a new session doesnt return an error I would assume my specific issue is fixed.

**Comment by MichaelS11** [comment]
@mattn Close this?
